Add Oracle Management Agent image (#2333)
oracle/docker-images: OracleManagementAgent/README.md

<!--
    Copyright (c) 2022, Oracle and/or its affiliates.
    Licensed under the Universal Permissive License v 1.0 as shown at
    https://oss.oracle.com/licenses/upl.
-->
Oracle Management Agent Container Image
=====
This repository contains sample container configurations to facilitate installation and environment setup for DevOps users. This project includes Dockerfiles based on Oracle Linux and Oracle OpenJDK 8.

The certification of Oracle Management Agent in a container does not require the use of any file presented in this
repository. Customers and users are welcome to use them as starters, and customize/tweak, or create
from scratch new scripts and Dockerfiles.

## How to build and run

Oracle Management Agent images use the Oracle Linux 7 Slim container image as the base image.

#### Build steps

1. [Download the Management Agent software](https://cloud.oracle.com/macs)

    **Note: Select 'Downloads and Keys' then download 'Agent for LINUX (X86_64)' of the package type ZIP.**

1. Copy the downloaded bundle to the same directory as the Dockerfile

    ```shell
    $ cp oracle.mgmt_agent.zip OracleManagementAgent/dockerfiles/latest/
    ```

> 1. Change directory to build the dockerfile

Reviewer: ```suggestion
1. Change to the directory in which the `Dockerfile` for this container image is located
```

Author: Fixed

    ```shell
    $ cd OracleManagementAgent/dockerfiles/latest/
    ```

> 1. Use Docker to build a container image

Reviewer: ```suggestion
1. Build the container image
```

Author: Fixed

    ```shell
    $ docker build -t oracle/mgmtagent-container .
    ```

> 1. [Create and Download the Install Key](https://docs.oracle.com/en-us/iaas/management-agents/doc/install-management-agent-chapter.html)
> 
>     **Note: Follow the steps in the 'Create Install Key' and 'Configure a Response File' sections to create the install key and save it as input.rsp.**

Reviewer: ```suggestion
1. Follow the steps in the [Create Install Key](https://docs.oracle.com/en-us/iaas/management-agents/doc/management-agents-administration-tasks.html#GUID-C841426A-2C32-4630-97B6-DF11F05D5712) and [Configure a Response File](https://docs.oracle.com/en-us/iaas/management-agents/doc/install-management-agent-chapter.html#GUID-5D20D4A7-616C-49EC-A994-DA383D172486) sections of the [Management Agent](https://docs.oracle.com/en-us/iaas/management-agents/index.html) documentation to create an install key and save it locally as `input.rsp`.
```

Author: Fixed

1. Create a Docker volume to share configs with the container

    ```shell
    $ docker volume create mgmtagent-volume

    # identify the mount point location to use in next steps
    $ docker volume inspect mgmtagent-volume|grep Mountpoint
        "Mountpoint": "/var/lib/docker/volumes/mgmtagent-volume/_data",
    ```

1. Copy the Install Key (input.rsp) into the shared Docker volume Mountpoint

    ```shell
    # create any necessary dirs
    $ mkdir -p /var/lib/docker/volumes/mgmtagent-volume/_data/mgmtagent_secret
    $ cp input.rsp /var/lib/docker/volumes/mgmtagent-volume/_data/mgmtagent_secret/

>     ```

Reviewer: Is the `input.rsp` required for first run or every run? This is not best practice when it comes to sharing content into a container. You should either use secrets (if the `input.rsp` contains credentials) or use a first-run script that obtains those secrets and stores them in the container.

Author: `input.rsp` is required up until the agent is registered. This is a runtime behavior dependent on what causes a registration failure, it can range from tenancy not configured correctly to any input misconfigured in the `input.rsp`. I take your point and have added docker-compose based container build and run execution. Docker-compose creates a secret and makes `input.rsp` available to the container without requiring additional user actions and hides the entire workflow.

> 1. [Apply Prerequisites for Deploying Management Agents](https://docs.oracle.com/en-us/iaas/management-agents/doc/perform-prerequisites-deploying-management-agents.html)
> 
>     **Note: Without the prerequisites applied the agent may not function as expected.**

Reviewer: ```suggestion
1. Ensure your tenancy is configured correctly by [applying the documented prerequisites for deploying management agents](https://docs.oracle.com/en-us/iaas/management-agents/doc/perform-prerequisites-deploying-management-agents.html)
```

Author: Fixed

> 1. Use Docker to start a container

Reviewer: ```suggestion
1. Start a container
```

Author: Fixed

    ```shell
    $ docker run -d --name mgmtagent-container --hostname mgmtagent1 -v mgmtagent-volume:/opt/oracle:rw --restart unless-stopped oracle/mgmtagent-container:latest
    ```

    **Description of Docker run parameters used above**
    <!-- markdownlint-disable MD033 -->
    | Parameter | Description |
    | --------- | ----------- |
    | -d | Starts mgmtagent-container in detached mode |
    | --name mgmtagent-container | The name given to the container to identify it. |
    | --hostname mgmtagent1 | Assign mgmtagent1 as the containers internal hostname. This can be any hostname compliant string and it will be used to identify the Management Agent instance in the OMC Console. |
    | -v mgmtagent-volume | /opt/oracle:rw: Mounts the volume mgmtagent-volume created on host filesystem inside the container at /opt/oracle with Read/Write privileges. |
    | --restart unless-stopped | Unless explicitly stopped, this restart policy restarts mgmtagent-container automatically when docker restarts. |
    <!-- markdownlint-enable MD033 -->

> #### Helpful Docker Administration commands

Reviewer: ```suggestion
#### Helpful administration commands
```

Author: Fixed

1. Starting a stopped Management Agent Container

    ```shell

>     $ docker container start mgmtagent-container

Reviewer: ```suggestion
    $ docker start mgmtagent-container
```

Author: Fixed

    ```

1. Stopping a running Management Agent Container

    ```shell

>     $ docker container stop mgmtagent-container

Reviewer: ```suggestion
    $ docker stop mgmtagent-container
```

Author: Fixed

    ```

1. Inspecting logs of Management Agent Container

    ```shell

>     $ docker container logs mgmtagent-container

Reviewer: ```suggestion
    $ docker logs mgmtagent-container
```

Author: Fixed

    ```

## License
To download and run the Oracle Management Agent, regardless whether inside or outside a container, you must download the binaries from the Oracle website and accept the license indicated at that page.

Oracle Linux is licensed under the [Oracle Linux End-User License Agreement](https://oss.oracle.com/ol/EULA).

All scripts and files hosted in this project and GitHub [`docker-images/OracleManagementAgent`](./) repository, required to build the Docker images are, unless otherwise noted, released under the [UPL 1.0](https://oss.oracle.com/licenses/upl/) license.

## Customer Support

> Oracle Management Agent container image is supported for Oracle Linux 7. For more details please see My Oracle Support.

Reviewer: I'm not sure I understand this line. Can I run this container only on Oracle Linux 7? Please provide a link to the specific My Oracle Support Article ID.

Author: Updated the doc and provided a link that better describes supported OS for Management Agent

<!--
    Copyright (c) 2022, Oracle and/or its affiliates.
    Licensed under the Universal Permissive License v 1.0 as shown at
    https://oss.oracle.com/licenses/upl.
-->
Oracle Management Agent on Docker
=====
This repository contains sample container configurations to facilitate installation and environment setup for DevOps users. This project includes Dockerfiles based on Oracle Linux and Oracle OpenJDK 8.

> The certification of Oracle Management Agent on Docker does not require the use of any file presented in this

Reviewer: Please replace "on Docker" with "in a container"

repository. Customers and users are welcome to use them as starters, and customize/tweak, or create
from scratch new scripts and Dockerfiles.

## How to build and run

> Oracle Management Agent images use the Oracle Linux 7 Slim docker image as the base image.

Reviewer: Please replace "docker" with "container"

Author: fixed

### Prerequisites

> * Latest Management Agent software for Linux (Linux-x86_64/latest/oracle.mgmt_agent.zip)

Author: fixed

#### Build steps

1. [Download the Management Agent software](https://cloud.oracle.com/macs)

    **Note: Select 'Downloads and Keys' then download 'Agent for LINUX (X86_64)' of the package type ZIP.**
    ```shell
    Copy-paste link: https://cloud.oracle.com/macs

>     ```

Reviewer: This is not a shell command.

1. Copy the downloaded bundle to the same directory as the Dockerfile

    ```shell
    $ cp oracle.mgmt_agent.zip OracleManagementAgent/dockerfiles/latest/
    ```

1. Change directory to build the dockerfile

    ```shell
    $ cd OracleManagementAgent/dockerfiles/latest/
    ```

> 1. Build the Docker image

Reviewer: Please rephrase this as "Use Docker to build a container image"

Author: fixed

    ```shell
    $ docker build -t oracle/mgmtagent-container .
    ```

1. [Create and Download the Install Key](https://docs.oracle.com/en-us/iaas/management-agents/doc/install-management-agent-chapter.html)

    **Note: Follow the steps in the 'Create Install Key' and 'Configure a Response File' sections to create the install key and save it as input.rsp.**
    ```shell
    $ docker volume create mgmtagent-volume

>     ```

Reviewer: This is a duplicate code block and should be removed.

1. Create a Docker volume to share configs with the container

    ```shell
    $ docker volume create mgmtagent-volume

    # identify the mount point location to use in next steps
    $ docker volume inspect mgmtagent-volume|grep Mountpoint
        "Mountpoint": "/var/lib/docker/volumes/mgmtagent-volume/_data",
    ```

1. Copy the Install Key (input.rsp) into the shared Docker volume Mountpoint

    ```shell
    # create any necessary dirs
    $ mkdir -p /var/lib/docker/volumes/mgmtagent-volume/_data/mgmtagent_secret
    $ cp input.rsp /var/lib/docker/volumes/mgmtagent-volume/_data/mgmtagent_secret/
    ```

1. [Apply Prerequisites for Deploying Management Agents](https://docs.oracle.com/en-us/iaas/management-agents/doc/perform-prerequisites-deploying-management-agents.html)

    **Note: Without the prerequisites applied the agent may not function as expected.**

> 1. Start the Docker container

Reviewer: Please rephrase this as "Use Docker to start a container"

Author: fixed

    ```shell
    $ docker run -d --name mgmtagent-container --hostname mgmtagent1 -v mgmtagent-volume:/opt/oracle:rw --restart unless-stopped oracle/mgmtagent-container:latest
    ```
    **Description of Docker run parameters used above**
    ```shell
    -d: Starts mgmtagent-container in detached mode
    --name mgmtagent-container: The name given to the container to identify it.
    --hostname mgmtagent1: Assign mgmtagent1 as the containers internal hostname. This can be any hostname compliant string and it will be used to identify the Management Agent instance in the OMC Console.
    -v mgmtagent-volume:/opt/oracle:rw: Mounts the volume mgmtagent-volume created on host filesystem inside the container at /opt/oracle with Read/Write privileges.
    --restart unless-stopped: Unless explicitly stopped, this restart policy restarts mgmtagent-container automatically when docker restarts.

>     ```

Author: fixed

#### Helpful Docker Administration commands

1. Starting a stopped Management Agent Container

    ```shell
    $ docker container start mgmtagent-container
    ```

1. Stopping a running Management Agent Container

    ```shell
    $ docker container stop mgmtagent-container
    ```

1. Inspecting logs of Management Agent Container

    ```shell
    $ docker container logs mgmtagent-container
    ```

## License
To download and run the Oracle Management Agent, regardless whether inside or outside a container, you must download the binaries from the Oracle website and accept the license indicated at that page.

Oracle Linux is licensed under the [Oracle Linux End-User License Agreement](https://oss.oracle.com/ol/EULA).

All scripts and files hosted in this project and GitHub [`docker-images/OracleManagementAgent`](./) repository, required to build the Docker images are, unless otherwise noted, released under the [UPL 1.0](https://oss.oracle.com/licenses/upl/) license.

## Customer Support

> Oracle Management Agent docker image is supported for Oracle Linux 7. For more details please see My Oracle Support.

Reviewer: Please replace "docker" with "container"

Author: fixed

<!--
    Copyright (c) 2022, Oracle and/or its affiliates.
    Licensed under the Universal Permissive License v 1.0 as shown at
    https://oss.oracle.com/licenses/upl.

> -->

Reviewer: Documentation files do not need copyright headers, so please remove this block.

Author: Fixed

> Oracle Management Agent Container Image
> =====

Reviewer: ```suggestion
# Oracle Management Agent Container Image

```

Author: Fixed

This repository contains sample container configurations to facilitate installation and environment setup for DevOps users. This project includes Dockerfiles based on Oracle Linux and Oracle OpenJDK 8.

The certification of Oracle Management Agent in a container does not require the use of any file presented in this
repository. Customers and users are welcome to use them as starters, and customize/tweak, or create
from scratch new scripts and Dockerfiles.

## How to build and run

> Oracle Management Agent images use the Oracle Linux 7 Slim container image as the base image.

Reviewer: ```suggestion
Oracle Management Agent images uses the official `oraclelinux:7-slim` container image as the base image. 
```

Reviewer: Though, I'm not entirely sure what the purpose of this sentence is. It doesn't provide any useful information that's not already contained in the `Dockerfile`.

Author: It's just an info statement that we found other docker-images projects had in their doc so we kept it thinking it was a convention
